packages/lisk-p2p/src/p2p.ts

```diff
@@ -12,59 +12,197 @@
+	public applyPenalty = (penalty: P2PPenalty): void => {
 		penalty;
 	};
 	// TODO
 	public getNetworkStatus = (): NetworkStatus => true;
 	// TODO
+
 	public getNodeStatus = (): P2PNodeStatus => true;
-	// TODO
-	public request = async (
-		packet: P2PRequestPacket,
-	): Promise<P2PResponsePacket> => {
-		const response = packet;
+	// TODO: Implement
 
-		return Promise.resolve(response);
-	};
+	public async request<T>(
+		packet: P2PRequestPacket<T>,
+	): Promise<P2PResponsePacket> {
+		return Promise.resolve({ data: packet });
+	}
 
-	public send = (message: P2PMessagePacket): void => {
+	public send<T>(message: P2PMessagePacket<T>): void {
 		message;
 		// TODO
-	};
+	}
 	public setNodeStatus = (nodeStatus: P2PNodeStatus): void => {
 		nodeStatus;
 		// TODO
 	};
-	// TODO
-	public start = async (): Promise<void> => this.peerPool.start();
-	// TODO
-	public stop = async (): Promise<void> => this.peerPool.stop();
+
+	private async _startPeerServer(): Promise<void> {
+		this._scServer.on(
+			'connection',
+			(socket: any): void => {
+				const queryObject = querystring.parse(socket.request.url);
+				if (
+					typeof queryObject.wsPort !== 'string' ||
+					typeof queryObject.os !== 'string' ||
+					typeof queryObject.version !== 'string'
+				) {
+					super.emit(EVENT_FAILED_TO_ADD_INBOUND_PEER);
+				} else {
+					const wsPort: number = parseInt(queryObject.wsPort, BASE_10_RADIX);
+					const peerId = P2P.generatePeerIdFromPeerInfo({
+						ipAddress: socket.remoteAddress,
+						wsPort,
+					});
+					const existingPeer = this._peerPool.getPeer(peerId);
+					if (existingPeer === undefined) {
+						const peer = new Peer({
+							height: 0,
+							id: peerId,
+							inboundSocket: socket,
+							ipAddress: socket.remoteAddress,
+							os: queryObject.os,
+							version: queryObject.version,
+							wsPort,
+							nodeInfo: this._nodeInfo,
+						});
+						this._peerPool.addPeer(peer);
+						super.emit(EVENT_NEW_INBOUND_PEER, peer);
+						super.emit(EVENT_NEW_PEER, peer);
+					} else {
+						existingPeer.inboundSocket = socket;
+					}
+					this._triedPeers.add(peerId);
+				}
+			},
+		);
+		this._httpServer.listen(this._config.wsPort);
+
+		return new Promise<void>(resolve => {
+			this._scServer.once('ready', () => {
+				resolve();
+			});
+		});
+	}
+
+	public async _loadListOfPeerListsFromSeeds(
+		seedList: ReadonlyArray<PeerInfo>,
+	): Promise<ReadonlyArray<ReadonlyArray<Peer>>> {
+		return Promise.all(
+			seedList.map(async (seedPeer: PeerInfo) => {
+				const seedPeerId = P2P.generatePeerIdFromPeerInfo(seedPeer);
+				const peer = new Peer({
+					id: seedPeerId,
+					ipAddress: seedPeer.ipAddress,
+					wsPort: seedPeer.wsPort,
+					nodeInfo: this._nodeInfo,
+				});
+				/**
+				 * TODO LATER: For the LIP phase, we shouldn't add the seed peers to our
+				 * _peerPool and we should disconnect from them as soon as we've loaded their peer lists.
+				 */
+				this._newPeers.add(seedPeerId);
```

Comment:
At that point, the seed peer has been created but has not been connected yet. It will be added to the triedPeers list after it connects (after request is called).